src/scenes/audio.js

```diff
@@ -73,10 +76,14 @@ export class AudioScene extends Phaser.Scene {
     // Pause BGM while playing SFX
     playSfx(audioName) {
         this.currentSong.pause();
-        let sfx = this.sound.add(audioName);
-        sfx.setVolume(this.currentVolume[SFX] / 4);
-        sfx.play();
-        sfx.once("complete", () => {
+        if (this.sfx != undefined && this.sfx.isPlaying) {
+            this.sfx.stop();
+            this.currentSong.pause();
+        }
+        this.sfx = this.sound.add(audioName);
+        this.sfx.setVolume(this.currentVolume[SFX] / 4);
+        this.sfx.play();
+        this.sfx.once("complete", () => {
             this.currentSong.resume();
         });
```

Comment:
Idk if these are the changes that caused this or if it was preexisting, but if you delete cookies, all subsequent volume settings will not be saved and it will always reset to 2/4 when switching scenes.

Reply:
Looks like it's pre-existing. I'll check it out

Reply:
I think it's a visual bug rather than an audio one

Reply:
It's both. I just fixed the visual issue of it, but the source still remains. It seems that when the cookies are reset the characterData pointer owned by everyone gets fractured. I can just put this.characterData = characterData at the top of every create function but this wouldn't work for the audio scene since it persists. I'll need to think of a way to propagate the change to everyone when cookies are reset. Either that or I'll need to make the characterData globally accessible through some getCharacterData() function.

Reply:
Ah, I think since the audio scene persists it doesn’t get the updated object and it keeps the old pointer

Reply:
Ah, I think since the audio scene persists it doesn’t get the updated object and it keeps the old pointer